includes/admin/promos/notices/class-license-upgrade-notice.php

```diff
@@ -0,0 +1,158 @@
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return false;
+		}
+
+		// Someone with no license keys entered always sees a notice.
+		if ( 0 === $this->number_license_keys ) {
+			return true;
+		}
+
+		// If we have no pass data yet, don't show the notice because we don't yet know what it should say.
+		if ( ! $this->pass_manager->has_pass_data ) {
+			return false;
+		}
+
+		// If someone has an extended pass or higher, and has an active AffiliateWP license, don't show.
+		try {
+			if (
+				$this->pass_manager->has_pass() &&
+				Pass_Manager::pass_compare( $this->pass_manager->highest_pass_id, Pass_Manager::EXTENDED_PASS_ID, '>=' ) &&
+				$this->has_affiliate_wp_license()
+			) {
+				return false;
+			}
+		} catch ( \Exception $e ) {
+			return true;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Determines whether or not AffiliateWP is installed and has a license key.
+	 *
+	 * @since 2.10.6
+	 *
+	 * @return bool
+	 */
+	private function has_affiliate_wp_license() {
+		if ( ! function_exists( 'affiliate_wp' ) ) {
+			return false;
+		}
+
+		return (bool) affiliate_wp()->settings->get( 'license_key' );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	protected function _display() {
+		try {
+			if ( 0 === $this->number_license_keys ) {
+
+				// No license keys active at all.
+				printf(
+				/* Translators: %s link URL */
+					__( 'You are using the free version of Easy Digital Downloads. <a href="%s" target="_blank">Purchase a pass</a> to get email marketing tools and recurring payments.', 'easy-digital-downloads' ),
+					add_query_arg( $this->query_args( 'core' ), 'https://easydigitaldownloads.com/pricing/' )
+				);
+
+			} elseif ( ! $this->pass_manager->highest_pass_id ) {
+
+				// Individual product license active, but no pass.
+				printf(
+				/* Translators: %s link URL */
+					__( 'For access to additional extensions to grow your store, consider <a href="%s" target="_blank">purchasing a pass</a>.', 'easy-digital-downloads' ),
+					add_query_arg( $this->query_args( 'extension-license' ), 'https://easydigitaldownloads.com/pricing/' )
+				);
+
+			} elseif ( Pass_Manager::pass_compare( $this->pass_manager->highest_pass_id, Pass_Manager::PERSONAL_PASS_ID, '=' ) ) {
+
+				// Personal pass active.
+				printf(
+				/* Translators: %s link URL */
+					__( 'You are using Easy Digital Downloads with a Personal pass. Consider <a href="%s" target="_blank">upgrading</a> to get recurring payments and more.', 'easy-digital-downloads' ),
+					add_query_arg( $this->query_args( 'personal-pass' ), 'https://easydigitaldownloads.com/your-account/license-keys/' )
+				);
+
+			} elseif ( Pass_Manager::pass_compare( $this->pass_manager->highest_pass_id, Pass_Manager::EXTENDED_PASS_ID, '>=' ) ) {
+
+				// Extended pass or higher.
+				printf(
+				/* Translators: %s link URL */
+					__( 'Grow your business and make more money with affiliate marketing. <a href="%s" target="_blank">Get AffiliateWP.</a>', 'easy-digital-downloads' ),
```

Comment:
Please remove the period (.) at the end of the linked text "Get AffiliateWP."

Reply:
Fixed here: https://github.com/easydigitaldownloads/easy-digital-downloads/pull/8679/commits/7d59aa452e2b0b95f1fb6d62e25d1ea71654cafe